### PR TITLE
Stream usage registry integration

### PR DESCRIPTION
**Pull request by jwang47**
Depends on https://github.com/caskdata/cdap/pull/2166

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/registry/UsageDatasetUtil.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.data2.dataset2.DatasetManagementException;
+import co.cask.cdap.proto.Id;
+
+import java.io.IOException;
+
+/**
+ * Utility to create {@link UsageDataset}.
+ */
+public class UsageDatasetUtil {
```
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/registry/UsageDatasetUtil.java]: Util class shouldn't be "newable" and should be final.

**Comment by jwang47** [comment]
Working on writing tests for this.

**Comment by anew** [comment]
At a quick glance, this seems a really large PR for as little as it is doing. Can you add a quick summary of what changes were made to help understand it?

**Comment by jwang47** [comment]
Most of it is from https://github.com/caskdata/cdap/pull/2166 (+541 −327), but I'll add a quick summary, yes.

**Comment by anew** [comment]
Ok, I just don't want to double-review 2166. I'll wait for 2166 and then review this.

**Comment by jwang47** [comment]
Reopend PR here https://github.com/caskdata/cdap/pull/2188

**Pull request closed without merging**